media/windows/wintasksmon/wintasksmon.py

```diff
@@ -0,0 +1,439 @@
+    """
+    json_tasks_run_state_list = []
+    # Attempt to open the tasks last-run-state file and read contents
+    try:
+        with open(ECCTasksLastState, "r") as tasks_run_states:
+            json_tasks_run_state_list = json.load(tasks_run_states)
+    # Handle [Errno 2] No such file or directory, JSON decoding error (syntax error in file)
+    except FileNotFoundError as e:
+        logger.info(F"No existing tasks last-run-state JSON file found...")
+        logger.info(F"...error:  {e}")
+        print(F"No existing tasks last-run-state JSON file found...")
+        print(F"...error:  {e}")
+    except json.JSONDecodeError as e:
+        logger.error(F"Invalid tasks last-run-state JSON file...aborting")
+        print(F"Invalid tasks last-run-state JSON file...aborting")
+        sys.exit(1)
+
+    return json_tasks_run_state_list
+
+
+def save_tasks_last_run_state(tasks_states):
+    """
+            This routine will save the list of tasks' last-states (task name, state).
+                    Written by DK Fowler ... 01-Nov-2021
+
+    :param tasks_states:     List of dictionaries containing the task name and last-run-state
+    :return:                True if successful, else False
+    """
+
+    try:
+        with open(ECCTasksLastState, "w") as tasks_run_states:
+            json.dump(tasks_states, tasks_run_states)
+    # Handle [Errno 2] No such file or directory, JSON decoding error (syntax error in file)
+    except json.JSONDecodeError as e:
+        logger.error(F"Invalid format for saving tasks last-run-states...{e}")
+        print(F"Invalid format for saving tasks last-run-states...{e}")
+        logger.error(F"...tasks run states contents:")
+        logger.error(F"......{tasks_states}")
+        print(F"...tasks run states contents:")
+        print(F"......{tasks_states}")
+        return False
+    except Exception as e:
+        logger.error(F"Error occurred while saving tasks last-run-states...{e}")
+        print(F"Error occurred while saving tasks last-run-states...{e}")
+        logger.error(F"...tasks run states contents:")
+        logger.error(F"......{tasks_states}")
+        print(F"...tasks run states contents:")
+        print(F"......{tasks_states}")
+        return False
+
+    return True
+
+
+def main():
+    logger.info(F"*** Initiating Epiphany Catholic Church Windows task monitor ***")
+    logger.info(F"Version:  {ecctasks_version}    Version date:  {ecctasks_date}")
+    json_task_list = get_tasks_to_monitor()  # get list of tasks and desired monitoring state
+    json_last_run_state_list = get_tasks_last_run_state()  # get list of tasks last-run-state
+    for task in json_task_list:
+        print(f'Task: {task["task"]}')
+        logger.debug(f'Task: {task["task"]}')
+    n = 0
+    for folder, subfolders, tasks in walk_tasks('/'):
+        n += len(tasks)
+        tasks_last_run_state = []  # create a list (of dictionaries) to hold current task, run-state
```

Comment:
I see in the loop below that you're appending to this list.  Are the `task_to_check["task"]` values unique?  If so, this could just be a dictionary indexed on `task_to_check["task"]`, and then you can just access the values directly (without needing to iterate through the list looking for the desired task).

Reply:
Yeah, that might be a bit more elegant / efficient...in this case it probably doesn't make much difference as there are probably only 1 (maybe 2) entries in the `task_to_check` AND `tasks_last_run_state` lists.

Reply:
FWIW: I'm not too worried about the efficiency -- you're absolutely correct that we're only talking about a handful of tasks, after all.  I was more thinking about the readability of the code; it's simpler this way (even if `tasks_to_check["task"]` ends up being being unique, and you have to put multiple things together into a single string to make a unique key).